api/rpm_and_image_fetcher.py

```diff
@@ -0,0 +1,72 @@
+import os
+import requests
+import logging
+import json
+
+# Set your GitHub access token as an environment variable
+GITHUB_TOKEN = os.environ.get("GITHUB_PERSONAL_ACCESS_TOKEN")
+headers = {
+    "Authorization": f"token {GITHUB_TOKEN}"
+}
+
+REPO_OWNER = "openshift-eng"
+REPO_NAME = "ocp-build-data"
+
+def get_directory_contents(branch, directory):
+    path = f"{REPO_OWNER}/{REPO_NAME}/contents/{directory}?ref={branch}"
+    contents_api = f"https://api.github.com/repos/{path}"
+    response = requests.get(contents_api, headers=headers)
+    response.raise_for_status()  # Raise an exception if the request fails
```

Comment:
I'm a bit puzzled about manually raising an exception here. If the get request fails, it'll itself raise a exception no?

Reply:
https://docs.python-requests.org/en/latest/user/quickstart/#response-status-codes
Idk i have seen it here (100% possible I could be wrong) , however since I have migrated to using GhApi, I dont think there is an need for it now from what I have seen.